### PR TITLE
Goodreads search functionalities

### PR DESCRIPTION
**Pull request by tahakucukkatirci**
I have finished the implementation of search_by_author function. Planning to finish the functions for genre and title asap.

### CODE REVIEW

File: bookbotproject/goodReadsApi.py [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
+	works = converted_response['GoodreadsResponse']['search']['results']['work']
+	for work in works:
+		book = work['best_book']['title']
+		books.append(book.encode('utf-8'))
 
-def search_by_author(intent):
-	str = "author"
-	book = gc.search_books(intent,str)
-	return search_books
-def search_by_genre(intent):
```
Comment (theTytoAlba) [inline bookbotproject/goodReadsApi.py:16]: Maybe do not delete these methods yet. We have the new search by author, we can delete the old one. But lets keep the other two until we have their new implementation. :)

**Comment by tahakucukkatirci** [comment]
I ended the implementation of search by genre and name too. It is ready to be merged.

**Comment by theTytoAlba** [comment]
Thank you! Merging!

**Comment by agoz17** [comment]
I have tested with some of our test cases and it works as it should be.  Thank you Taha this was vital.

**Pull request merged**